### DISCUSSION TITLE
How to get a list of columns and measures with their descriptions for documentation purposes?

### DISCUSSION THREAD

**Discussion by jdayton-maker** [DaxStudio/DaxStudio#535] [Q&A]
I'm using Power Pivot for my model and dashboards.  Unlike Power BI, when you float over a column / measure in Power Pivot, it does NOT show the underlying description for that column / measure.  In daxstudio I can get lists of columns with DBSCHEMA_COLUMNS and measures with MDSCHEMA_MEASURES but even though there is a Description column, it is not populated.  Is there a way I can get all this information in one fell swoop?  I'm not ready to tackle MDX or Power Query just yet as digesting DAX was more than enough for now. 

Any help would be appreciated.

**Comment by dgosbell** [comment]
So I can't see the description  in DBSCHEMA_COLUMNS either, but I'm not sure if that is a bug or just a limitation of that DMV. For Power BI or later versions of SSAS you could use the TMSCHEMA_COLUMNS view, but that is not available for PowerPivot. One work around for PowerPivot is to use the MDSCHEMA_LEVELS view. Each column gets an auto-generated attribute hierarchy for backward compatibility with MDX client tools, so if you exclude user defined hierarchies and exclude the "All" level you effectively get a row per column. I added a description to my country column in the example below and you can see that is visible.

```
SELECT [DIMENSION_UNIQUE_NAME] as TABLE_NAME
	,[LEVEL_NAME] as COLUMN_NAME
	,[DESCRIPTION]
FROM $SYSTEM.MDSCHEMA_LEVELS
WHERE [LEVEL_ORIGIN] <> 1 -- excludes User Defined Hierarchies
  AND [LEVEL_NUMBER] = 1  -- excludes the (All) level
  AND [CUBE_NAME] = 'Model'
```

![image](https://user-images.githubusercontent.com/8241830/106372889-74da3a80-63c8-11eb-879a-692a498a7d01.png)

However I did not have any issues with MDSCHEMA_MEASURES, I added a description to Measure1 and that was reflected in that DMV.
 
```
SELECT [MEASURE_NAME], [DESCRIPTION] 
FROM $SYSTEM.MDSCHEMA_MEASURES
```

![image](https://user-images.githubusercontent.com/8241830/106372901-8de2eb80-63c8-11eb-82c3-98f032e00ca7.png)

**Comment by jdayton-maker** [comment]
Darren-

With the addition of one column to the latter sql, I got exactly what I wanted from your two queries:

SELECT [MeasureGroup_Name], [MEASURE_NAME], [DESCRIPTION] 
FROM $SYSTEM.MDSCHEMA_MEASURES

Thanks so much for your help!!!!
